Realm.Shared/Realm.cs

```diff
@@ -432,27 +429,27 @@ internal RealmObject MakeObjectForRow(string className, RowHandle rowHandle)
         internal ResultsHandle MakeResultsForTable(string className)
         {
             var metadata = Metadata[className];
-            IntPtr resultsPtr = NativeResults.create_for_table(SharedRealmHandle, metadata.Table, metadata.Schema.Handle);
+            var resultsPtr = NativeTable.CreateResults(metadata.Table, SharedRealmHandle, metadata.Schema.Handle);
             return CreateResultsHandle(resultsPtr);
         }
 
 
         internal ResultsHandle MakeResultsForQuery(Schema.ObjectSchema schema, QueryHandle builtQuery, SortOrderHandle optionalSortOrder)
         {
-            IntPtr resultsPtr = IntPtr.Zero;               
+            var resultsPtr = IntPtr.Zero;               
             if (optionalSortOrder == null)
-                resultsPtr = NativeResults.create_for_query(SharedRealmHandle, builtQuery, schema.Handle);
+                resultsPtr = builtQuery.CreateResults(SharedRealmHandle, schema.Handle);
             else
-                resultsPtr = NativeResults.create_for_query_sorted(SharedRealmHandle, builtQuery, schema.Handle, optionalSortOrder);
+                resultsPtr = builtQuery.CreateSortedResults(SharedRealmHandle, schema.Handle, optionalSortOrder);
             return CreateResultsHandle(resultsPtr);
         }
 
 
         internal SortOrderHandle MakeSortOrderForTable(string className)
         {
-            var tableHandle = Metadata[className].Table;
-            IntPtr sortOrderPtr = NativeSortOrder.create_for_table(tableHandle);
-            return CreateSortOrderHandle(sortOrderPtr);
+            var result = new SortOrderHandle();
```

Comment:
Very true. There is an additional refactoring that we could do regarding the construction of handles all over. They are wrapped in `ConstrainedExecutionRegion` and generally aren't too intuitive. I just felt that it might be too radical a change in one go, so I kept the old style more or less.